client_test.go

```diff
@@ -0,0 +1,164 @@
+package helmclient
+
+import (
+	"helm.sh/helm/v3/pkg/repo"
+	"k8s.io/client-go/rest"
+)
+
+func ExampleNew() {
+	opt := &Options{
+		RepositoryCache:  "/tmp/.helmcache",
+		RepositoryConfig: "/tmp/.helmrepo",
+		Debug:            true,
+		Linting:          true,
+	}
+
+	// Construct a new Helm client, where '_' is the constructed client.
+	// Change this empty assignment to get access to the client's services.
+	_, err := New(opt)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleNewClientFromRestConf() {
+	opt := &RestConfClientOptions{
+		Options: &Options{
+			RepositoryCache:  "/tmp/.helmcache",
+			RepositoryConfig: "/tmp/.helmrepo",
+			Debug:            true,
+			Linting:          true,
+		},
+		RestConfig: &rest.Config{},
+	}
+
+	// Construct a new Helm client via REST configuration, where '_' is the constructed client.
+	// Change this empty assignment to get access to the client's services.
+	_, err := NewClientFromRestConf(opt)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleNewClientFromKubeConf() {
+	opt := &KubeConfClientOptions{
+		Options: &Options{
+			RepositoryCache:  "/tmp/.helmcache",
+			RepositoryConfig: "/tmp/.helmrepo",
+			Debug:            true,
+			Linting:          true,
+		},
+		KubeContext: "",
+		KubeConfig:  []byte{},
+	}
+
+	// Construct a new Helm client via KubeConf, where '_' is the constructed client.
+	// Change this empty assignment to get access to the client's services.
+	_, err := NewClientFromKubeConf(opt)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleHelmClient_AddOrUpdateChartRepo_public() {
+	// Dummy assignment
+	// Construct a real Helm client via New(), NewClientFromRestConf(), or NewClientFromKubeConf()
+	helmClient := &HelmClient{}
```

Comment:
Possible workaround for all the dummy assignments: Declare a `var helmClient *HelmClient` in another file in this package (maybe `doc.go` or similar) and just use that. It doesn't need to be initialized, just needs to be there.